bridge-stp.in

```diff
@@ -59,27 +114,65 @@ checkpid()
 
 case $2 in
     start)
-        if [ -e "$conf_file" ]; then
-            . "$conf_file" || exit
+        # Make sure the specified bridge is valid.
+        if [ ! -d "$net_dir/$bridge/bridge" ]; then
+            echo "'$bridge' is not a bridge" >&2
+            exit 1
+        fi
+
+        # Determine whether the specified bridge should use MSTP.
+        if ! is_mstp_bridge "$bridge"; then
+            echo "Ignoring bridge '$bridge' that is not listed in \$MSTP_BRIDGES"
+            exit 10
```

Comment:
The kernel only cares whether the exit status is zero or non-zero, so `exit 1`, `exit 2`, `exit 10`, etc are all treated the same by the kernel.  However, for debugging purposes, newer kernels will print the exit status of the script so the user can figure out what is going on.

In this case, if a bridge is missing from `$MSTP_BRIDGES`, that's not really an "error" ... it is a valid configuration that is intentionally causing a non-zero error status to be returned to the kernel so that this bridge will use kernel STP instead of mstpd.  In that case, I want the exit status printed by the kernel to be clearly different than the exit status which is printed when there is a real error.  Hence why I used 1/2/3 for real errors and 10 for this "error".  There is nothing special about 10 except that it is obviously different than 1/2/3.